parser/parser.go

```diff
@@ -83,19 +90,14 @@ type Program struct {
 	// but are exported for the interpreter (Program itself needs to
 	// be exported in package "parser", otherwise these could live in
 	// "internal/ast".)
-	Begin     []ast.Stmts
-	Actions   []ast.Action
-	End       []ast.Stmts
-	Functions []ast.Function
-	Scalars   map[string]int
-	Arrays    map[string]int
-	Compiled  *compiler.Program
+	ast.ResolvedProgram
```

Comment:
At first I was a little hesitant to change this (as `Program` is exported), but I think I've convinced myself it's okay, especially given the above comment. Plus, the fields are still accessible (via the embedded `ResolvedProgram` struct) if anyone really is using them (they shouldn't be).

Reply:
Ok so I assume we don't need to change anything here. Resolving then without changes. Please let me know if you think otherwise.